pkg/infrastructure/baremetal/bootstrap.go

```diff
@@ -451,9 +459,22 @@ func destroyBootstrap(config baremetalConfig) error {
 	}
 
 	logrus.Debug("  Undefining domain")
-	err = virConn.DomainUndefine(dom)
-	if err != nil {
-		return err
+
+	if err := virConn.DomainUndefineFlags(dom, libvirt.DomainUndefineNvram|libvirt.DomainUndefineSnapshotsMetadata|libvirt.DomainUndefineManagedSave|libvirt.DomainUndefineCheckpointsMetadata); err != nil {
+		var libvirtErr *libvirt.Error
+
+		if !errors.As(err, &libvirtErr) {
+			return errors.New("failed to cast to libvirt.Error")
```

Comment:
I don't know if we expect this to ever happen, but maybe use `fmt.Errorf` with %w to show the wrapped error?

Reply:
Done